GeneratorInterface/ExternalDecays/src/ExternalDecayDriver.cc

```diff
@@ -24,61 +24,51 @@ CLHEP::HepRandomEngine* decayRandomEngine = nullptr;
     }
     else if( curSet == "EvtGen1" || curSet == "EvtGen130" ) {
-      fEvtGenInterface = (EvtGenInterfaceBase*)(EvtGenFactory::get()->create("EvtGen130", pset.getUntrackedParameter< ParameterSet >(curSet)));
+      fEvtGenInterface = std::unique_ptr<EvtGenInterfaceBase>(EvtGenFactory::get()->create("EvtGen130", pset.getUntrackedParameter< ParameterSet >(curSet)));
       exSharedResources.emplace_back(edm::SharedResourceNames::kEvtGen);
       exSharedResources.emplace_back(edm::SharedResourceNames::kPythia8);
       exSharedResources.emplace_back(edm::SharedResourceNames::kTauola);
       exSharedResources.emplace_back(edm::SharedResourceNames::kPhotos);
       exSharedResources.emplace_back(gen::FortranInstance::kFortranInstance);
     }
     else if ( curSet == "Tauola" || curSet == "Tauolapp" || curSet == "Tauolapp114" ) {
-      fTauolaInterface = (TauolaInterfaceBase*)(TauolaFactory::get()->create("Tauolapp114", pset.getUntrackedParameter< ParameterSet >(curSet)));
-      fPhotosInterface = (PhotosInterfaceBase*)(PhotosFactory::get()->create("Photos2155", pset.getUntrackedParameter< ParameterSet >(curSet)));
+      fTauolaInterface = std::unique_ptr<TauolaInterfaceBase>(TauolaFactory::get()->create("Tauolapp114", pset.getUntrackedParameter< ParameterSet >(curSet)));
+      fPhotosInterface = std::unique_ptr<PhotosInterfaceBase>(PhotosFactory::get()->create("Photos2155", pset.getUntrackedParameter< ParameterSet >(curSet)));
       fPhotosInterface->configureOnlyFor( 15 );
       fPhotosInterface->avoidTauLeptonicDecays();
       exSharedResources.emplace_back(edm::SharedResourceNames::kTauola);
       exSharedResources.emplace_back(edm::SharedResourceNames::kPhotos);
     }
     else if ( curSet == "Photos" || curSet == "Photos2155" ) {
       if ( !fPhotosInterface ) {
-	fPhotosInterface = (PhotosInterfaceBase*)(PhotosFactory::get()->create("Photos2155", pset.getUntrackedParameter< ParameterSet>(curSet)));
+	fPhotosInterface = std::unique_ptr<PhotosInterfaceBase>(PhotosFactory::get()->create("Photos2155", pset.getUntrackedParameter< ParameterSet>(curSet)));
 	exSharedResources.emplace_back(edm::SharedResourceNames::kPhotos);
       }
     }
     else if (curSet == "Photospp" || curSet == "Photospp356" ) {
       if ( !fPhotosInterface ) {
-        fPhotosInterface = (PhotosInterfaceBase*)(PhotosFactory::get()->create("Photospp356", pset.getUntrackedParameter< ParameterSet>(curSet)));
+        fPhotosInterface = std::unique_ptr<PhotosInterfaceBase>(PhotosFactory::get()->create("Photospp356", pset.getUntrackedParameter< ParameterSet>(curSet)));
         exSharedResources.emplace_back(edm::SharedResourceNames::kPhotos);
       }
     }
   }
 }
 
-
-ExternalDecayDriver::~ExternalDecayDriver()
-{
-  if ( fEvtGenInterface ) delete fEvtGenInterface;
-  if ( fTauolaInterface ) delete fTauolaInterface;
-  if ( fPhotosInterface ) delete fPhotosInterface;
-}
-
+ExternalDecayDriver::~ExternalDecayDriver() = default;
```

Comment:
@makortel in principle this could just be declared in the header at this point? Anyway ok for me

Reply:
Only moving the destructor definition to header would not be enough as also the includes of `TauolaInterfaceBase.h` etc would have to be moved (in order for the compiler to generate the call to their destructor). That in turn *might* increase physical coupling of the clients of `ExternalDecayDriver` to the decayers.

Reply:
@makortel ok I see...